src/core/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,233 @@
+    hourStep = 1,
+    minuteStep = 1,
+    secondStep = 1,
+  } = props;
+
+  useTheme();
+
+  const [selectedTime, setSelectedTime] = React.useState(date);
+
+  React.useEffect(() => {
+    setSelectedTime(date);
+  }, [date]);
+
+  const onHourClick = (hour: number) => {
+    const adjustedSelectedDay = selectedTime ?? new Date();
+    adjustedSelectedDay.setHours(hour);
+    updateCurrentTime(adjustedSelectedDay);
+  };
+
+  const onMinutesClick = (minutes: number) => {
+    const adjustedSelectedDay = selectedTime ?? new Date();
+    adjustedSelectedDay.setMinutes(minutes);
+    updateCurrentTime(adjustedSelectedDay);
+  };
+
+  const onSecondsClick = (seconds: number) => {
+    const adjustedSelectedDay = selectedTime ?? new Date();
+    adjustedSelectedDay.setSeconds(seconds);
+    updateCurrentTime(adjustedSelectedDay);
+  };
+
+  const updateCurrentTime = (time: Date) => {
+    setSelectedTime(time);
+    onChange?.(time);
+  };
+
+  // const handleTimeKeyDown = (
+  //   event: React.KeyboardEvent<HTMLLIElement>,
+  //   maxValue: number,
+  //   callback: (value: number) => void,
+  //   onClick: (value: number) => void,
+  //   currentValue?: number,
+  // ) => {
+  //   if (!focusedTime || currentValue == null) {
+  //     return;
+  //   }
+  //   switch (event.key) {
+  //     case 'ArrowDown':
+  //       if (currentValue + 1 > maxValue) {
+  //         break;
+  //       }
+  //       callback(currentValue + 1);
+  //       needFocus.current = true;
+  //       event.preventDefault();
+  //       break;
+  //     case 'ArrowUp':
+  //       if (currentValue - 1 < 0) {
+  //         break;
+  //       }
+  //       callback(currentValue - 1);
+  //       needFocus.current = true;
+  //       event.preventDefault();
+  //       break;
+  //     case 'Enter':
+  //     case ' ':
+  //     case 'Spacebar':
+  //       onClick(currentValue);
+  //       event.preventDefault();
+  //       break;
+  //   }
+  // };
+
+  const scrollIntoView = (ref: HTMLLIElement | null, isSame: boolean) => {
+    isSame && ref?.scrollIntoView({ block: 'nearest', inline: 'nearest' });
+  };
+
+  const getTimeTemplate = (
+    data: number[],
+    step: number,
+    isSame: (number: number, date: Date | undefined) => boolean,
+    onClick: (number: number) => void,
+  ) => {
+    return (
+      <div className='iui-time'>
+        <ol>
+          {data
+            .filter((_, index) => index % step === 0)
+            .map((value) => {
+              return (
+                <li
+                  className={cx({
+                    'iui-selected': isSame(value, selectedTime),
+                  })}
+                  key={value}
+                  tabIndex={isSame(value, selectedTime) ? 0 : undefined}
+                  ref={(ref) => {
+                    scrollIntoView(ref, isSame(value, selectedTime));
+                  }}
+                  onClick={() => onClick(value)}
+                >
+                  {`0${value}`.slice(-2)}
```

Comment:
This is interesting 🤔 Could it cause problems anywhere?

Reply:
I guess in future we should provide ability to format the shown value.. Just dont want to add all the features at once here, it is really endless.

Reply:
Now I have valueRenderer prop, which we could expose to the users and they could then show what ever they want.

Reply:
How is the valueRenderer (part of `TimePickerColumn`) exposed to the user? I'm confused

Reply:
It is not yet, but it might be.. Just need to think of API.. Multiple props (hour, second, etc..) or something else..

Reply:
Added new props for every column cell value. So now it can be customized by the user, what we gonna show in there (including AM and PM for localization)